### PR TITLE
Reworks Cults stun hand

### PR DESCRIPTION
**Pull request by hal9000PR**
## What Does This PR Do
title.
using the stun hand one someone will apply a mark, cause them to be knocked down, silence, and do 60 stamina damage.
hitting the target again with a cult dagger, sword or weilded cult spear will detonate the mark, fully stunning them and refreshing the silence.
also adds an attack animation to cult stun.
hitting them with the sword/dagger does do damage, however the convert rune heals damage.
the mark lasts as long as the knockdown duration.

## Why It's Good For The Game
less instant stuns. more counterplay.  my one concern is this might mean that cults die out earlier and faster. TG tried reworking the cult stun many a time, so this is something I am wary of.

## Images of skrell getting owned
needs a screen alert and maybe a new effect if I can get a hold of one.
dreamseeker_WZVfb0kXiE

## Changelog
:cl:
tweak: reworked cult stun hand. now knocks down, causes stam damage and applies a mark, instead of stunning. hitting the target again with a cult dagger or sword will fully stun them and refresh the silence.
/:cl:

### CODE REVIEW

**Comment by Miraviel** [comment]
This looks cool. :o

When you stab them with the sword and drag them away, do they leave behind a blood trail? (If yes, that'll interfere with stealth converting a lot.)

**Comment by Leanfrog** [comment]
> This looks cool. :o
> 
> When you stab them with the sword and drag them away, do they leave behind a blood trail? (If yes, that'll interfere with stealth converting a lot.)
> 
> edit: not sword, the dagger

Depends on how much damage they take, you see in the GIF provided the dagger doesn't immediately create a pool on the floor while the sword equivalent does. Additionally blood rites exists and can be used to clean up a trail you don't want existing.

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -149,6 +149,29 @@
...
+	var/mob/living/carbon/human/H = owner
+	H.add_overlay(overlay)
+
+/datum/status_effect/cult_stun_mark/on_remove()
+	owner.cut_overlay(overlay)
+
+/datum/status_effect/cult_stun_mark/proc/trigger()
+	owner.adjustStaminaLoss(60)
+	owner.Silence(6 SECONDS) // refresh the silence
+	to_chat(world, "cowabunga")
```
Comment (Qwertytoforty) [inline code/datums/status_effects/debuffs.dm:172]: I too have everyone in the world hear cowabunga when I stab someone with a sword

Comment (Qwertytoforty) [inline code/datums/status_effects/debuffs.dm:172]: I too have everyone in the world hear cowabunga when I wack a person with a sword

**Review by Qwertytoforty — COMMENTED** [review]
Will want to change this debug message

File: code/datums/status_effects/debuffs.dm (hunk 1 above)
Comment (Sirryan2002) [reply code/datums/status_effects/debuffs.dm:172]: no no, charlie is readding soul to the game

File: code/game/gamemodes/cult/cult_items.dm [hunk 2]
```diff
@@ -46,6 +46,10 @@
 		else
 			user.adjustBruteLoss(rand(force/2, force))
 		return
+	if(!iscultist(target))
```
Comment (KalevTait) [inline code/game/gamemodes/cult/cult_items.dm:49]: Is this check neccessary? It means that cultist friendly fire is off, and maybe that's desireable... but maybe it's not?

**Comment by matttheficus** [comment]
Would it be worth making so you could double Stun spell someone to proc the mark in the events you want zero Cult items on yourself?

**Comment by Qwertytoforty** [comment]
> Would it be worth making so you could double Stun spell someone to proc the mark in the events you want zero Cult items on yourself?

A second cult spell will send the stamina damage over 100 (directly to 120), stunning them

**Comment by KittyNoodle** [comment]
My worry is that this will increase the cult skill gap between people who can swap between hands fast and grab a dagger (or know to do that in the first place) and those who don’t.

**Comment by Dragonkiller93** [comment]
From my understanding, unless there's a delay between stun hand and dagger built into the code, this would be a near instant stun, plus knockdown meaning near impossible to beat. I don't see much of a difference between this and old stun hand for experienced players, but do correct me if I'm wrong.

**Comment by hal9000PR** [comment]
> this will increase the cult skill gap

Agreed. However I don't have any solution to this.

**Comment by Sirryan2002** [comment]
code looks good

File: code/datums/status_effects/debuffs.dm [hunk 3]
```diff
@@ -178,6 +178,29 @@
 /datum/status_effect/pacifism/on_remove()
 	REMOVE_TRAIT(owner, TRAIT_PACIFISM, id)
 
+// used to track if hitting someone with a cult dagger/sword should stamina crit.
+/datum/status_effect/cult_stun_mark
+	id = "cult_stun"
+	duration = 10 SECONDS // the when they stand back up, it will be over
```
Comment (SteelSlayer) [inline code/datums/status_effects/debuffs.dm:184]: Comment grammar :)

File: code/datums/status_effects/debuffs.dm [hunk 4]
```diff
@@ -178,6 +178,29 @@
...
+	alert_type = null
+	var/mutable_appearance/overlay
+
+/datum/status_effect/cult_stun_mark/on_apply()
+	. = ..()
+	if(!. || !ishuman(owner))
+		return
```
Comment (SteelSlayer) [inline code/datums/status_effects/debuffs.dm:191]: Currently the parent doesn't do anything so I would remove the `. = ..()` and `!.`

File: code/game/gamemodes/cult/cult_items.dm [hunk 5]
```diff
@@ -538,7 +542,7 @@
 
 /obj/item/twohanded/cult_spear
 	name = "blood halberd"
-	desc = "A sickening spear composed entirely of crystallized blood."
+	desc = "A sickening spear composed entirely of crystallized blood. Will stun people who have been recently marked if the spear is weilded."
```
Comment (SteelSlayer) [inline code/game/gamemodes/cult/cult_items.dm:545]:
```suggestion
	desc = "A sickening spear composed entirely of crystallized blood. Will stun people who have been recently marked if the spear is wielded."
```

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**